ls1intum/Artemis
# [Feature] Show success message for late submissions and results

### Checklist

### Motivation and Context


If the exercise name has `BASE` or `SOLUTION` in its name, the student participation cannot be found during the result creation.
There is no indication for a student if he submits a programming exercise after the due date. So we want to have a success message for that.
Attempt to solve #2915 (?)

### Description

This fixes the bug with searching for a participation during the result creation. We now check if the build plan key ends with the string `-BASE` or `-SOLUTION` instead of just checking if the name exists within the string.

When a submission has no result and we are triggering a failed build within the Course Exercise Row or Course Exercise Detail, we are not using `lastGraded = true` anymore and are searching for any submission not only graded ones.

### Steps for Testing

#### Test 1
1. Log in to Artemis
2. Participate in an exercise where the due date is over
3. Make changes and click on `Submit`
4. Directly afterwards change to the Course Exercise Detail and verify a green success notification which states that a late submission was received
5. Wait a few seconds and verify that another success notification shows that an `ungraded` result was received
6. At last the result history on the bottom of the page should have been updated with the latest result

#### Test 2
1. Log in to Artemis
2. Create a programming exercise where the name contains either BASE or SOLUTION
3. Participate as a student and verify that the build result is received as normal

### Test Coverage



course-exercise-details.component.ts: 80%

### Screenshots
